src/status_im/mailserver/core.cljs

```diff
@@ -504,22 +507,27 @@
   (when-not (zero? (:peers-count db))
     (if-let [preferred-mailserver (preferred-mailserver-id db)]
       (let [current-fleet (node/current-fleet-key db)]
+        ;; Error connecting to the mail server
         {:db
-         (update-mailserver-state db :error)
-         :ui/show-confirmation
-         {:title               (i18n/label :t/mailserver-error-title)
-          :content             (i18n/label :t/mailserver-error-content)
-          :confirm-button-text (i18n/label :t/mailserver-pick-another)
-          :on-accept           #(re-frame/dispatch
-                                 [:navigate-to (if platform/desktop?
-                                                 :advanced-settings
-                                                 :offline-messaging-settings)])
-          :extra-options       [{:text    (i18n/label :t/mailserver-retry)
-                                 :onPress #(re-frame/dispatch
-                                            [:mailserver.ui/connect-confirmed
-                                             current-fleet
-                                             preferred-mailserver])
-                                 :style   "default"}]}})
+         (update-mailserver-state db :error)}
+        (let [error-dismissed (connection-error-dismissed db)]
+          ;; Only show the error popup if it was not dismissed previously
+          (when-not error-dismissed
+            {:ui/show-confirmation
+             {:title               (i18n/label :t/mailserver-error-title)
+              :content             (i18n/label :t/mailserver-error-content)
+              :confirm-button-text (i18n/label :t/mailserver-pick-another)
+              :on-cancel           #(re-frame/dispatch [:mailserver.ui/dismiss-connection-error true])
+              :on-accept           #(re-frame/dispatch
+                                      [:navigate-to (if platform/desktop?
+                                                      :advanced-settings
+                                                      :offline-messaging-settings)])
+              :extra-options       [{:text    (i18n/label :t/mailserver-retry)
+                                     :onPress #(re-frame/dispatch
```

Comment:
This is unrelated to the changes in this PR, but I'm wondering if there's a reason `onPress` is written in camelCase here while most other events seem to be using kebab-case

Reply:
kebab-case should be used

Reply:
This was code that was already there, I just changed the indentation. I can change it to kebab-case if it's an available prop though.

Reply:
yeah, as i said its a really old code, should be re-implemented from scratch